<?xml version="1.0"?>
<launch>
  <arg name="bag_path" />
  <arg name="map_path" />
  <arg name="rviz" default="true"/>
  <arg name="robot_description" default="$(find xacro)/xacro --inorder $(find drz_telemax_description)/urdf/drz_telemax_complete.urdf.xacro"/>

  <!-- Load robot description -->
  <param name="robot_description" command="$(arg robot_description)" />
  <param name="robot_description_semantic" textfile="$(find drz_telemax_moveit_config)/config/drz_telemax.srdf" />

  <node pkg="hector_pose_prediction_benchmark" type="benchmark_node" name="pose_prediction_benchmark" output="screen" clear_params="true" ><!--launch-prefix="gdb -args"-->
    <param name="pose_predictor" value="hector_heightmap_pose_prediction"/>
    <param name="bag_file_path" value="$(arg bag_path)"/>
    <param name="result_folder" value="$(find hector_pose_prediction_benchmark)/data"/>
    <param name="path_sampling_resolution" value="0.05"/>
    <param name="wait_time" value="0.0"/>

    <rosparam subst_value="true" ns="hector_heightmap_pose_prediction/heightmap">
      map_bag_file_path: $(arg map_path)
    </rosparam>
  </node>

  <!-- rviz ui -->
  <include if="$(arg rviz)" file="$(find hector_pose_prediction_benchmark)/launch/rviz.launch"/>
</launch>
